main.py

```diff
@@ -261,14 +261,24 @@ def _kill_tb_child():
     atexit.register(_kill_tb_child)
 
 
-def get_best_checkpoint_path(run_dir, phase):
+def get_best_checkpoint_path(run_dir, phase, task_name=None):
     """ Look in run_dir for model checkpoint to load.
     Hierarchy is
-        1) best checkpoint from the phase so far
-        2) if we do only target training without pretraining, then load checkpoint before 
+        1) best task-specific checkpoint
+        2) best checkpoint from the phase so far
+        3) if we do only target training without pretraining, then load checkpoint before 
         target training
-        3) nothing found (empty string) """
-    checkpoint = glob.glob(os.path.join(run_dir, "model_state_%s_epoch_*.best_macro.th" % phase))
+        4) nothing found (empty string) 
+    """
+    checkpoint = []
+    if task_name is not None:
+        checkpoint = glob.glob(
+            os.path.join(run_dir + "/" + task_name + "/", "model_state_%s_epoch_*.best.th" % phase)
+        )
+    if len(checkpoint) == 0:
+        checkpoint = glob.glob(
+            os.path.join(run_dir, "model_state_%s_epoch_*.best_macro.th" % phase)
```

Comment:
Make a note that we'll only see these for 'target_task' when loading older runs.

Reply:
In this PR, we still have checkpoints named model_state_target_train_epoch_{}{}.best_macro.th, that store the checkpoints of models that maximize the macro_avg (which might be different than the best checkpoints for each task).

Reply:
Ah, that's broken, then—we can't and shouldn't save the best checkpoints for the individual tasks during pretraining. (They're not guaranteed to be good 'bests', since we might early stop before some tasks are well trained.) Just track a single 'best' in pretraining.